ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js

```diff
@@ -1135,8 +1135,13 @@ if (typeof window.DatePickerWidget === 'undefined') {
      */
 
     setValue(value) {
-      let currDate =  new Date(value);
-
+      let currDate;
+      const dateArray = value.split('-');
+      if (dateArray && dateArray.length == 3) {
+        currDate = new Date(dateArray[0], dateArray[1] - 1, dateArray[2]);
```

Comment:
For all date parsing logic, we should rely on OOTB formatting logic which is exposed in af-formatters. @vdua Why can't af-core handle this ?

Reply:
+1 needs to be handled in af2-web-runtime

Reply:
The issue is specific to the view layer, @barshat7 is working on the fix.